packages/web/src/hooks/swap/data/use-swap.tsx

```diff
@@ -50,7 +61,15 @@ export const useSwap = ({ tokenA, tokenB, direction, slippage, swapFee = 15 }: U
       tokenAmount: direction === "EXACT_IN" ? swapAmount : swapAmount ? swapAmount * exactOutPadding : swapAmount,
     },
     {
-      enabled: !!swapAmount && swapAmount > 0 && !!tokenA && !!tokenA.path && !!tokenB && !!tokenB.path && !isSameToken,
+      enabled:
+        shouldFetch &&
+        !!swapAmount &&
+        swapAmount > 0 &&
+        !!tokenA &&
+        !!tokenA.path &&
+        !!tokenB &&
+        !!tokenB.path &&
+        !isSameToken,
```

Comment:
Although `!!` expression is idiomatic in TypeScript, in this case it would be better to create a local variable for each logical check. This makes it clear what each condition represents and its intended purpose.

like:

```typescript
const hasValidSwapAmount = swapAmount && swapAmount > 0;
const hasValidTokenPaths = Boolean(tokenA?.path) && Boolean(tokenB?.path);
const isDifferentTokens = !isSameToken;

const isEnabled = shouldFetch && hasValidSwapAmount && hasValidTokenPaths && isDifferentTokens;
```

Reply:
For clarity of purpose, we managed the logic checks as local variables.